contents/scalability/scalable-development.md

```diff
@@ -29,26 +50,42 @@ In certain scenarios, the runtime complexity can be reduced as a trade-off betwe
+The next thing to consider is what to cache. The rule of thumb is to cache data that is frequently accessed and read more often than it is updated. With this, cache hits would be more often than cache misses and as a result, the time saved from faster accesses and reads outweighs the extra time taken to populate the cache.  
 
-Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases/)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+#### Sharding 
+As your traffic increases your data increases as well and as a result, your database gets overloaded. One way to mitigate this is to scale your database by sharding. Sharding is a method of splitting and storing a single logical dataset in multiple databases. More specifically, it is the storing of data horizontally - storing rows of a same table in multiple database nodes instead of storing them in the usual vertical way - storing different tables & columns in a separate database.
 
-### Get More Machines
+What are the benefits of Sharding?
+* Sharding can store more data than a single database can hold -
+Sharding is essential when your dataset becomes too large to store in a single database. It reduces the number of rows in each table and as such improves search performance since the search is done on a smaller table. 
 
-Buy more machines to host your server. This is called horizontal scaling. This method can help improve scalability by a huge factor, as each additional unit of processing power and RAM costs the same, no matter how many machines you buy. As this is a promising way to increase scalability, it is no surprise why large services such as Google and Facebook use thousands of computer servers just to provide services to all their users.
+* Sharding can make queries faster - Querying the databases containing only the relevant partitions becomes possible as well. For instance, if a database contains a column for age, you can partition the rows according to an age group and store them in different databases. Whenever there is a need to access the data of a particular age group, instead of querying the whole database, you just need to query the partition that contains that age group. This allows your database to scale along with your data and traffic growth.
 
-However, to fully reap the benefits of multiple machines, you will need to learn various techniques to modify your server infrastructure. In most cases you cannot simply run multiple instances of the server and expect it to work.
+#### Go Asynchronous 
+Unlike synchronous operations, which runs sequentially and waits for the previous operation to complete before moving on to the next, asynchronous operations are non-blocking i.e. does not block further execution until that operation finishes. As a result other operations do not have to wait for asynchronous operations to finish.
```

Comment:
> i.e. does not block further execution until that operation finishes.

Quite hard to understand. I think it's unnecessary for this to be here, since you already have a "As a result..." to explain what asynchronous operations mean.

Reply:
`i.e. does not block further execution until that operation finishes`
Oh, this was to explain the term non-blocking as mentioned by Prof. So it's better to replace `non-blocking` with  `does not block further execution until that operation finishes` right?